consensus/scheme/rolldpos/fsm.go

```diff
@@ -476,9 +476,17 @@ func (m *cFSM) handleProposeBlockEvt(evt fsm.Event) (fsm.State, error) {
 		if err != nil {
 			return sInvalid, errors.Wrap(err, "error when calculating the proposer")
 		}
+		producer := proposeBlkEvt.block.ProducerAddress()
+		if producer == "" {
+			return sInvalid, errors.New("Invalid producer address when validating proposer")
```

Comment:
Do not throw error. It's not a system issue, but an invalid block. This validator should pre-vote "no"